org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/FieldTableInspector.java

```diff
@@ -0,0 +1,116 @@
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.openjdk.jmc.console.ext.agent.manager.wizards;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jface.viewers.ColumnLabelProvider;
+import org.eclipse.jface.viewers.ColumnViewerToolTipSupport;
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Tree;
+import org.openjdk.jmc.console.ext.agent.manager.model.IField;
+import org.openjdk.jmc.ui.column.ColumnBuilder;
+import org.openjdk.jmc.ui.column.ColumnManager;
+import org.openjdk.jmc.ui.column.IColumn;
+import org.openjdk.jmc.ui.misc.OptimisticComparator;
+import org.openjdk.jmc.ui.misc.TreeStructureContentProvider;
+
+public class FieldTableInspector {
+
+	public static final String FIELD_TREE_NAME = "EventEditingWizardFieldPage.FieldsTree"; //$NON-NLS-1$
+
+	private final TreeViewer viewer;
+
+	private final ColumnLabelProvider nameLabelProvider = new ColumnLabelProvider() {
+		@Override
+		public String getText(Object element) {
+			if (element instanceof IField) {
+				return ((IField) element).getName();
+			}
+			return element.toString();
+		}
+	};
+
+	private final ColumnLabelProvider expressionLabelProvider = new ColumnLabelProvider() {
+		@Override
+		public String getText(Object element) {
+			if (element instanceof IField) {
+				return ((IField) element).getExpression();
+			}
+			return element.toString();
+		}
+	};
+
+	private final ColumnLabelProvider descriptionLabelProvider = new ColumnLabelProvider() {
+		@Override
+		public String getText(Object element) {
+			if (element instanceof IField) {
+				return ((IField) element).getDescription();
+			}
+			return element.toString();
+		}
+	};
+
+	public FieldTableInspector(Composite parent) {
+		Tree tree = new Tree(parent,
+				SWT.BORDER | SWT.FULL_SELECTION | SWT.MULTI | SWT.VIRTUAL | SWT.H_SCROLL | SWT.V_SCROLL);
+		tree.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		viewer = new TreeViewer(tree);
+		viewer.setContentProvider(new TreeStructureContentProvider());
+		ColumnViewerToolTipSupport.enableFor(viewer);
+
+		tree.setData("name", FIELD_TREE_NAME); //$NON-NLS-1$
```

Comment:
what's the purpose of this?

Reply:
I think its if you want to access the tree another time, the name identifies it.  Im not sure if we need it in our context though.

Reply:
After a further look at the `MBeanTreeSectionPart`, where I believe you borrowed this idea from, it seems this `.setData` the call is solely for UI testing purpose. 

We don't really have any UI tests at this time. While I'm not strongly against having this line, I'll leave this one for you to decide. You can either remove it or resolve this conversation yourself. :)

Reply:
Oh i saw this is already removed. nvm